api/src/main/java/org/apache/iceberg/ChangeScan.java

```diff
@@ -0,0 +1,104 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg;
+
+import java.util.List;
+
+/**
+ * API for configuring a table scan for changes.
+ */
+public interface ChangeScan extends Scan<TableScan> {
+  /**
+   * Instructs this scan to look for changes starting from a particular snapshot (inclusive).
+   * <p>
+   * If the start snapshot is not configured, it is defaulted to the oldest ancestor
+   * of the end snapshot (inclusive).
+   *
+   * @param fromSnapshotId the start snapshot ID (inclusive)
+   * @return this for method chaining
+   */
+  ChangeScan fromSnapshotInclusive(long fromSnapshotId);
+
+  /**
+   * Instructs this scan to look for changes starting from a particular snapshot (exclusive).
+   * <p>
+   * If the start snapshot is not configured, it is defaulted to the oldest ancestor
+   * of the end snapshot (inclusive).
+   *
+   * @param fromSnapshotId the start snapshot ID (exclusive)
+   * @return this for method chaining
+   */
+  ChangeScan fromSnapshotExclusive(long fromSnapshotId);
+
+  /**
+   * Instructs this scan to look for changes up to a particular snapshot (inclusive).
+   * <p>
+   * If the end snapshot is not configured, it is defaulted to the current table snapshot (inclusive).
+   *
+   * @param toSnapshotId the end snapshot ID (inclusive)
+   * @return this for method chaining
+   */
+  ChangeScan toSnapshot(long toSnapshotId);
+
+  /**
+   * Instructs this scan to assign equality delete files to matching data files,
+   * which may be expensive if equality deletes are not scoped to particular partitions.
+   *
+   * @return this for method chaining
+   */
+  ChangeScan resolveEqualityDeletes();
+
+  /**
+   * Plans scan tasks for all file changes.
+   *
+   * @return this for method chaining
+   */
+  FileChangeIndex planFiles();
+
+  /**
+   * An index for looking up file scan tasks for changes in particular snapshots.
+   */
+  interface FileChangeIndex {
+    /**
+     * Returns a list of snapshots that are covered by this index.
+     */
+    List<Snapshot> snapshots();
+
+    /**
+     * Returns scan tasks for data files added in a given snapshot.
+     */
+    Iterable<FileScanTask> addedDataFiles(long snapshotId);
+
+    /**
+     * Returns scan tasks for data files removed in a given snapshot.
+     */
+    Iterable<FileScanTask> removedDataFiles(long snapshotId);
+
+    /**
+     * Returns scan tasks for data files that have at least one matching delete file added in a given snapshot.
+     */
+    Iterable<FileScanTask> dataFilesWithDeletes(long snapshotId);
+
+    /**
+     * Returns scan tasks for equality delete files that were not assigned to matching data files in a given snapshot.
+     */
+    Iterable<FileScanTask> unresolvedEqualityDeleteFiles(long snapshotId);
```

Comment:
IIUC, these are purely eq delete files. We may just call it `equalityDeleteFiles`?

Reply:
The idea is that this method would only return equality deletes not assigned to data files (if `resolveEqualityDeletes` is not called). If we assign equality deletes, then this will be empty.

Reply:
Resolving as no longer applies.